src/plugins/query_enhancements/server/connections/clients/base_connection_client.ts

```diff
@@ -12,11 +12,7 @@ export interface ResourcesQuery {
   query: Record<string, string>;
 }
 
-export interface ClientRequest {
-  query: ResourcesQuery;
-  path: string;
-  params: Partial<TransportRequestParams>;
-}
+export type ClientRequest = Omit<TransportRequestParams, 'method'>;
```

Comment:
It looks like this will no longer vend our ResourceQuery types. Can you do a union of these types so we can still have strong typing for consumers?

ex.
```
export type ClientRequest = Omit<TransportRequestParams, 'method'> & {
  query: ResourceQuery;
}
```

Reply:
i think the issue is `ClientRequest.params` is supposed to be passed to `transport.request()` call, but i missed the `.params` in design

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/6fa0a60eea464322c5118940ed529cc5c06ae388/src/plugins/query_enhancements/server/connections/clients/sql_connection_client.ts#L28
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/6fa0a60eea464322c5118940ed529cc5c06ae388/src/plugins/query_enhancements/server/connections/clients/promql_connection_client.ts#L28

if these two are `...request.params` instead of `...request` then we shouldn't need to change prometheus_manager nor the type definition for `ClientRequest`